sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/Optimizer.scala

```diff
@@ -769,7 +769,9 @@ object LimitPushDown extends Rule[LogicalPlan] {
       LocalLimit(exp, project.copy(child = pushLocalLimitThroughJoin(exp, join)))
     // Push down limit 1 through Aggregate and turn Aggregate into Project if it is group only.
     case Limit(le @ IntegerLiteral(1), a: Aggregate) if a.groupOnly =>
-      Limit(le, Project(a.aggregateExpressions, LocalLimit(le, a.child)))
+      val project = Project(a.aggregateExpressions, LocalLimit(le, a.child))
+      project.setTagValue(Project.dataOrderIrrelevantTag, ())
```

Comment:
According to the `EliminateSorts`, it's data order relevant if only group expressions.

Reply:
it's irrelevant, see https://github.com/apache/spark/blob/8d64cb4144d17107fd3758d2a46430439203c7ad/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/Optimizer.scala#L1629-L1647

Reply:
It seems `isOrderIrrelevantAggFunction` returns false if only group expressions.

Reply:
I missing the `checkValidAggregateExpression`.